### PR TITLE
Adding workflow for pypi publishing

### PR DESCRIPTION
**Pull request by amitschang**

### CODE REVIEW

**Comment by amitschang** [comment]
I'm pretty confident in this now, as I just went ahead and made a test pre-release: https://github.com/ssec-jhu/dplutils/actions/runs/6724499944.

File: .github/workflows/dist.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: DIST
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
```
Comment (jamienoss) [inline .github/workflows/dist.yml:10]: I don't think we want to publish everything, just releases, right?
```suggestion
on:
  release:
    types: [ released, prereleased ]
```
With ``prereleased`` being optional. See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

**Comment by jamienoss** [comment]
@amitschang thanks for adding this, we should also add it to https://github.com/ssec-jhu/base-template. Once merged we can trivially cherry-pick it over. However, I think there we'd def only want it to run on releases or not at all, or perhaps ``on: workflow_dispatch`` with the release code commented out.

**Comment by amitschang** [comment]
@jamienoss , makes sense. I added your change, and will do the cherry pick over to base-templates after we merge this.

**Review by jamienoss — APPROVED** [review]

**Pull request merged**